### PR TITLE
Removed simple kc client

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

File: cmd/mbop/main.go [hunk 1]
```diff
@@ -186,23 +215,35 @@ func getUser(w http.ResponseWriter, r *http.Request) (*User, error) {
...
-	_, err = k.GetGenericToken("redhat-external", username, password)
+	oauthClientConfig := clientcredentials.Config{
+		ClientID:       "admin-cli",
+		ClientSecret:   "",
+		TokenURL:       KEYCLOAK_SERVER + "auth/realms/redhat-external/protocol/openid-connect/token",
+		EndpointParams: url.Values{"grant_type": {"password"}, "username": {username}, "password": {password}},
+	}
+
+	k := oauthClientConfig.Client(context.Background())
+	resp, err := k.Get(fmt.Sprintf("%sauth/realms/redhat-external/account/", KEYCLOAK_SERVER))
```
Comment (Victoremepunto) [inline cmd/mbop/main.go:226]: this requires a / , the KEYCLOAK_SERVER is not expected to have a trailing `/' .

```suggestion
	resp, err := k.Get(fmt.Sprintf("%s/auth/realms/redhat-external/account/", KEYCLOAK_SERVER))
```

File: cmd/mbop/main.go [hunk 2]
```diff
@@ -186,23 +215,35 @@ func getUser(w http.ResponseWriter, r *http.Request) (*User, error) {
 		return &User{}, fmt.Errorf("can't create keycloak client: %s", err.Error())
 	}
 
-	_, err = k.GetGenericToken("redhat-external", username, password)
+	oauthClientConfig := clientcredentials.Config{
+		ClientID:       "admin-cli",
+		ClientSecret:   "",
+		TokenURL:       KEYCLOAK_SERVER + "auth/realms/redhat-external/protocol/openid-connect/token",
```
Comment (Victoremepunto) [inline cmd/mbop/main.go:221]: this requires a `/`  , the `KEYCLOAK_SERVER` is not expected to have a trailing `/' . 

```suggestion
		TokenURL:       KEYCLOAK_SERVER + "/auth/realms/redhat-external/protocol/openid-connect/token",
```

File: cmd/mbop/main.go [hunk 3]
```diff
@@ -276,8 +317,8 @@ type usersSpec struct {
 	Attributes map[string][]string `json:"attributes"`
 }
 
-func getUsers() (users []User, err error) {
-	resp, err := k.Get("/admin/realms/redhat-external/users?max=2000", "", map[string]string{})
+func (m *MBOPServer) getUsers() (users []User, err error) {
+	resp, err := m.Client.Get(fmt.Sprintf("%sauth/admin/realms/redhat-external/users?max=2000", KEYCLOAK_SERVER))
```
Comment (Victoremepunto) [inline cmd/mbop/main.go:321]: this requires a `/`  , the `KEYCLOAK_SERVER` is not expected to have a trailing `/' . 

```suggestion
	resp, err := m.Client.Get(fmt.Sprintf("%s/auth/admin/realms/redhat-external/users?max=2000", KEYCLOAK_SERVER))
```

File: cmd/mbop/main.go [hunk 4]
```diff
@@ -465,68 +510,95 @@ func entitlements(w http.ResponseWriter, r *http.Request) {
...
 	log = zapr.NewLogger(zapLog)
 
-	key, err := keycloak.NewKeyCloakClient(KEYCLOAK_SERVER, KEYCLOAK_USERNAME, KEYCLOAK_PASSWORD, context.Background(), "master", log, KEYCLOAK_VERSION)
-
-	if err != nil {
-		panic(err)
+	oauthClientConfig := clientcredentials.Config{
+		ClientID:       "admin-cli",
+		ClientSecret:   "",
+		TokenURL:       KEYCLOAK_SERVER + "auth/realms/master/protocol/openid-connect/token",
```
Comment (Victoremepunto) [inline cmd/mbop/main.go:547]: this requires a `/`  , the `KEYCLOAK_SERVER` is not expected to have a trailing `/' . 

```suggestion
		TokenURL:       KEYCLOAK_SERVER + "/auth/realms/master/protocol/openid-connect/token",
```

**Review by Victoremepunto — DISMISSED** [review]
I spotted what the problem was, left you some comments. Please review

File: build/Dockerfile [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # Build the manager binary
-FROM registry.access.redhat.com/ubi8/go-toolset:1.15.7 as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:1.17.12 as builder
 
 WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
-# COPY go.sum go.sum
+COPY go.sum go.sum
```
Comment (Victoremepunto) [inline build/Dockerfile:7]: Why wasn't this needed before ? just for my understanding, please.

File: cmd/mbop/main.go [hunk 6]
```diff
@@ -7,15 +7,16 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"net/url"
 	"os"
 	"sort"
 	"strconv"
 	"strings"
 
-	keycloak "github.com/RedHatInsights/simple-kc-client"
```
Comment (Victoremepunto) [inline cmd/mbop/main.go:15]: it's great that we can kill off dependencies, thank you!

**Review by Victoremepunto — COMMENTED** [review]
I'd ask some more experienced Go developer provide a review, looks fine, at least tests should be passing

File: build/Dockerfile (hunk 5 above)
Comment (psav) [reply build/Dockerfile:7]: We moved to go 1.17 I think that's why

**Pull request merged**